components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/MobileNumberVerificationHandler.java

```diff
@@ -95,10 +99,16 @@ public void handleEvent(Event event) throws IdentityEventException {
             }
             /* We need to empty 'MOBILE_NUMBER_PENDING_VALUE_CLAIM' because having a value in that claim implies
             a verification is pending. But verification is not enabled anymore. */
-            if (claims.containsKey(IdentityRecoveryConstants.MOBILE_NUMBER_CLAIM)) {
+            if (claims.containsKey(IdentityRecoveryConstants.MOBILE_NUMBER_CLAIM) ||
+                    claims.containsKey(IdentityRecoveryConstants.VERIFIED_MOBILE_NUMBERS_CLAIM)) {
```

Comment:
This is not required as we are checking whether verified mobile numbers available separately.